src/device-registry/utils/bulk-insert-measurements.js

```diff
@@ -5,9 +5,7 @@ const { logObject } = require("../utils/log");
 
 const insertDeviceMeasurements = {
 
-    // Contains modules that provide various ways to insert device measurements.
-    
-    addValuesArray: async (array_data) => {
+    addValuesArray: async (measurements_array) => {
```

Comment:
Not sure if it is connected to this but recently realised that the endpoint for inserting many measurements can actually insert values for different devices at the same time....the logic in the code accepts that....the device query parameter provided on the endpoint is secondary in priority so if any measurement in the array has a device attribute, that takes precedence over the query parameter.... :)

Reply:
Yes it is connected, But it also depends on the flow of code execution while inserting measurements. Revisit the following files....You might need to make updates where necessary
https://github.com/airqo-platform/AirQo-api/blob/staging/src/device-registry/controllers/create-event.js
https://github.com/airqo-platform/AirQo-api/blob/staging/src/device-registry/utils/transform-measurements.js
https://github.com/airqo-platform/AirQo-api/blob/staging/src/device-registry/utils/insert-measurements.js

Reply:
Cool, thanks for the clarification...